kata.py

```diff
@@ -1,8 +1,29 @@
 def Add(string):
-    if string == "":
+    def Empty(string):
+        if string == "":
+            return True
+    if Empty(string):
         return 0
     else:
-        answer = 0
-        for num in string.replace('\n',',').split(','):
-            answer += int(num)
-        return answer 
\ No newline at end of file
+        if string[:2] == '//':
+            if string.count('\n') > 1:
+                if Empty(string[4:]):
+                    return 0
+                else:
+                    answer = 0
+                    for num in string[4:].split('\n'):
+                        answer += int(num)
+                    return answer
+            elif Empty(string.split('\n')[1]):
+                return 0
+            else:
+                answer = 0
+                delimiter = string.split('\n')[0].strip('//')
+                for num in string.split('\n')[1].split(delimiter):
+                    answer += int(num)
+                return answer
```

Comment:
Nested `if` conditionals are often a great indicator that your function is starting to do many things - which can create confusion, and code that's very hard to test and change (the Gilded Rose is perhaps the idiomatic example of that). This is probably the best/first opportunity in the kata to really think about the 'refactor' stage of red/green/refactor - notice how your original code just sort of stays sitting there quite merrily at the end of all those if/elses, with a little extra indentation.

Might your function now have multiple responsibilities? `Add` is doing quite a bit more than just adding - and while we're constrained a little by the fixed method signature perhaps we can extract some additional functions out here to keep everything having one single responsibility.

I think it's probably fair to say that in an ideal world our code in this function wouldn't get much more complicated than the original contents in that final `else` block.